jmespath-core/src/main/java/io/burt/jmespath/jcf/JcfAdapter.java

```diff
@@ -143,6 +154,15 @@ public Object getProperty(Object value, String name) {
   }
 
   @Override
+  public Collection<String> getPropertyNames(Object value) {
+    if (isObject(value)) {
+      return ((Map<String, Object>) value).keySet();
```

Comment:
I'm thinking that you at some point feed a context object into the evaluator, and if that context object has maps with non-string keys, "strange things" (tm) will happen. I wouldn't mind a `ClassCastException`, but no such thing will be raised here, as the cast is only a `Map` (without parameters) at runtime.